java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java

```diff
@@ -57,6 +59,11 @@ public void scanFile(JavaFileScannerContext context) {
   }
 
   @Override
+  public void visitEnumConstant(EnumConstantTree tree) {
+    scan(((NewClassTree) tree.initializer()).classBody());
```

Comment:
I created ticket and fixed it : https://github.com/SonarSource/sonar-java/pull/291